src/runtime-rs/crates/resource/src/cpu_mem/swap.rs

```diff
@@ -0,0 +1,537 @@
+            } else {
+                return Err(anyhow!(
+                    "swap_task: device_info {} pci_path is None",
+                    swap_path
+                ));
+            };
+
+            if let Err(e) = ret.await {
+                if let Err(e1) = self
+                    .device_manager
+                    .write()
+                    .await
+                    .try_remove_device(&device_id)
+                    .await
+                {
+                    error!(
+                        sl!(),
+                        "swap_task: try_remove_device {} fail: {:?}", swap_path, e1
+                    );
+                }
+
+                return Err(anyhow!("swap_task: agent.swap_add failed: {:?}", e));
+            }
+        } else {
+            return Err(anyhow!("swap_task: device_info {} is not Block", swap_path));
+        }
+
+        let mut core_lock = self.core.lock().await;
+        core_lock.update_next_swap_id();
+        core_lock.plus_swap_size(swap_size);
+
+        info!(
+            sl!(),
+            "swap_task: swap file {:?} {} inserted", swap_path, swap_size
+        );
+
+        Ok(())
+    }
+
+    async fn create_disk(&mut self, swap_size: usize, swap_path: &PathBuf) -> Result<()> {
+        debug!(
+            sl!(),
+            "swap_task: create_disk {:?} {} start", swap_path, swap_size
+        );
+
+        check_disk_size(&self.path, swap_size)
+            .await
+            .context("check_disk_size")?;
+
+        debug!(
+            sl!(),
+            "swap_task: create swap file {:?} {}", swap_path, swap_size
+        );
+        let mut file = File::create(swap_path)
+            .await
+            .context(format!("swap: File::create {:?}", swap_path))?;
+        fs::set_permissions(swap_path, Permissions::from_mode(0o700))
+            .await
+            .context(format!("swap: File::set_permissions {:?}", swap_path))?;
+
+        let buffer = vec![0; CHUNK_SIZE];
+        let mut total_written = 0;
+        while total_written < swap_size {
+            spawn_blocking(sched_yield)
+                .await
+                .context("swap_task: task::spawn_blocking")?
+                .context("swap_task: sched_yield")?;
+
+            if self.should_stop(false).await {
+                return Err(StStop::get_error(true));
+            }
+
+            let remaining = swap_size - total_written;
+            let write_size = std::cmp::min(remaining, CHUNK_SIZE);
+            file.write_all(&buffer[..write_size])
+                .await
+                .context("file.write_all")?;
+            total_written += write_size;
+        }
+
+        if self.should_stop(false).await {
+            return Err(StStop::get_error(true));
+        }
+
+        file.flush().await.context("file.flush")?;
+        drop(file);
+
+        if self.should_stop(false).await {
+            return Err(StStop::get_error(true));
+        }
+
+        let output = Command::new("mkswap")
```

Comment:
Does the binary of `mkswap` always exist ? If not exist, how to handle this error case ?